src/bridgetree.rs

```diff
@@ -295,32 +306,40 @@ pub struct Frontier<H, const DEPTH: u8> {
 }
 
 impl<H, const DEPTH: u8> Frontier<H, DEPTH> {
-    /// Construct a new empty frontier.
-    pub fn new() -> Self {
+    /// Constructs a new empty frontier.
+    pub fn empty() -> Self {
         Frontier { frontier: None }
     }
 
-    /// Return the position of latest leaf appended to the frontier,
+    /// Constructs a new frontier from a `[NonEmptyFrontier]`
+    ///
+    /// Returns `None` if the provided frontier exceeds the maximum
+    /// allowed depth.
+    pub fn new(frontier: NonEmptyFrontier<H>) -> Option<Self> {
+        if frontier.size() >= 1 << DEPTH {
```

Comment:
Double-check this edge case; I think this might actually need to be:
```suggestion
        if frontier.size() > 1 << DEPTH {
```